src/common/parallel_for.cpp

```diff
@@ -61,13 +65,29 @@ void ParallelFor::par_for_inc(const bitCapInt begin, const bitCapInt itemCount,
             futures[cpu].get();
         }
     } else {
+#if ENABLE_UINT128
```

Comment:
Use some macro's to set this up so that the code itself doesn't have ENABLE_UINT128 inline.

I'd recommend DECLARE_ATOMIC_BITCAPINT, ATOMIC_ASYNC(...), and ATOMIC_INC().